dev/lint-python

```diff
@@ -82,6 +82,25 @@ else
     rm "$PYCODESTYLE_REPORT_PATH"
 fi
 
+# stop the build if there are Python syntax errors or undefined names
+flake8 . --count --select=E901,E999,F821,F822,F823 --show-source --statistics
```

Comment:
Are we asking this as required in local too? Or we shall skip this if it is not installed on local?

Reply:
My take is that we want contributors to be able to locally detect syntax errors and undefined names _before_ they submit a pull request.

Reply:
If so, maybe it's still good to detect `flake8` and show some message to ask for install.

Reply:
I think flake8 not being found is a pretty clear message you need to install flake8. If we list it in `dev/requirements.txt` maybe that would help folks?